### PR TITLE
Allow multiple entries in EMU_PLATFORM

### PR DESCRIPTION
**Pull request by fkokosinski**
This PR introduces a way to define the `EMU_PLATFORM` variable as a list, and to choose which emulation platform the simulation will be run on by either setting the `EMU_PLATFORM_DEFAULT` variable in the per-board cmake file, or by using the added command line argument, e.g.:

```
west build -p always -b hifive1 -e renode -t run samples/hello_world
```

If `EMU_PLATFORM_DEFAULT` and command line argument have not been set, then it defaults to running the simulation on the first emulation platform from `EMU_PLATFORM` list.

### CODE REVIEW

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
-1 to the west build changes.

- There isn't sufficient justification in the commit log for why the EMU_PLATFORM_TARGET variable is special enough to deserve its own option. There are dozens of CMake variables of importance to users, and wrapping them all in west build options is simply not practical. Any exceptions need to have a clear justification.
- Even if this were justified appropriately, this would belong in a separate commit in my opinion.

**Comment by fkokosinski** [comment]
Thanks for your comment @mbolivar-nordic!

> There are dozens of CMake variables of importance to users, and wrapping them all in west build options is simply not practical. Any exceptions need to have a clear justification

I imagined that it would be beneficial enough for an individual user to be able to select the emulation platform directly while using `west build` command to justify the changes in west, but if that's not the case, I can drop this change.

**Comment by tejlmand** [comment]
> > There are dozens of CMake variables of importance to users, and wrapping them all in west build options is simply not practical. Any exceptions need to have a clear justification
> 
> I imagined that it would be beneficial enough for an individual user to be able to select the emulation platform directly while using `west build` command to justify the changes in west, but if that's not the case, I can drop this change.

what @mbolivar-nordic refers to is the fact that you can already do:
```
west build ... -- -DEMU_PLATFORM_TARGET=<emu_platform>
```
so there need to be a very good reason why the emu platform deserves its own west build argument.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1588,7 +1588,19 @@ if(HEX_FILES_TO_MERGE)
 endif()
 
 if(EMU_PLATFORM)
-  include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
+  if(NOT EMU_PLATFORM_TARGET)
+    if(NOT EMU_PLATFORM_DEFAULT)
+      list(GET EMU_PLATFORM 0 EMU_PLATFORM_DEFAULT)
+      message(WARNING "EMU_PLATFORM_DEFAULT is not set! Defaulting to "${EMU_PLATFORM_DEFAULT})
```
Comment (tejlmand) [inline CMakeLists.txt:1594]: No warning here, users should be able to build without warnings if they just do `west build` or `cmake ...`

You can propose a status informing the user which emu platform has been selected, but i'm not convinced that even such a message should be printed at all.

There are boards where people are building for real hardware that still defines a `EMU_PLATFORM` and telling those users about the emu platform is just noise. 

Also it could be confusing to see such message in the cases where they actually intend to flash the board and will not be using an emulator.
https://docs.zephyrproject.org/latest/boards/arm/mps2_an385/doc/index.html
https://github.com/zephyrproject-rtos/zephyr/blob/3053a931a31b0f72a7b149edf8ed2a244f8273a9/boards/arm/mps2_an385/board.cmake#L3

File: boards/riscv/hifive1/board.cmake [hunk 2]
```diff
@@ -1,9 +1,11 @@
 # SPDX-License-Identifier: Apache-2.0
 
-set(EMU_PLATFORM qemu)
+set(EMU_PLATFORM renode qemu)
+set(EMU_PLATFORM_DEFAULT renode)
```
Comment (tejlmand) [inline boards/riscv/hifive1/board.cmake:4]: let's avoid the `EMU_PLATFORM_DEFAULT` variable.
If `EMU_PLATFORM` is a list, then the first entry in that list should be considered the default emulation platform.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1588,7 +1588,19 @@ if(HEX_FILES_TO_MERGE)
...
+    endif()
+    set(EMU_PLATFORM_TARGET ${EMU_PLATFORM_DEFAULT})
+  endif()
+
+  if(EXISTS "${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM_TARGET}.cmake")
```
Comment (tejlmand) [inline CMakeLists.txt:1599]: To me, this looks like the `EMU_PLATFORM_TARGET` will always be used respectively of whether the `EMU_PLATFORM` list contains that platform or not.

For example:
```
# Some board.cmake
set(EMU_PLATFORM qemu renode)
```
User executes:
```
cmake ... -DEMU_PLATFORM_TARGET=tsim
 ```
then this code
```
 include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM_TARGET}.cmake)
```
will execute as:
```
 include(${ZEPHYR_BASE}/cmake/emu/tsim.cmake)
```
because tsim.cmake exists, is that intentional ?

File: scripts/pylib/twister/twisterlib.py [hunk 4]
```diff
@@ -2017,6 +2017,7 @@ def run_cmake(self, args=[]):
         cmake_args = [
             f'-B{self.build_dir}',
             f'-S{self.source_dir}',
+            f'-DEMU_PLATFORM_TARGET={self.platform.simulation}',
```
Comment (tejlmand) [inline scripts/pylib/twister/twisterlib.py:2020]: I don't think this is needed in twister.
```suggestion
```

Extra arguments to CMake can already be provided like this:
https://github.com/zephyrproject-rtos/zephyr/blob/3053a931a31b0f72a7b149edf8ed2a244f8273a9/tests/misc/test_build/testcase.yaml#L5

**Review by tejlmand — CHANGES_REQUESTED** [review]
Some general comments.

Also, please add appropriate documentation for this feature.

**Comment by tejlmand** [comment]
as I value useability for end-user over internal variable names, then we could consider to rename existing `EMU_PLATFORM` to something else, like `SUPPORTED_EMU_PLATFORMS` as it would become a list instead, and not necessarily the platform in use.

The `EMU_PLATFORM` is then freed up as user argument.
A `board.cmake` supporting only a single platform may continue to use `EMU_PLATFORM` as that is the only valid value, and hence there is no reason for the user to specify something else.

**Comment by mbolivar-nordic** [comment]
>  @mbolivar-nordic refers to is the fact that you can already do:
> 
> ```
> west build ... -- -DEMU_PLATFORM_TARGET=<emu_platform>
> ```
> 
> so there need to be a very good reason why the emu platform deserves its own west build argument.

Yes, exactly.



> I imagined that it would be beneficial enough for an individual user to be able to select the emulation platform directly while using `west build` command to justify the changes in west, but if that's not the case, I can drop this change.

As shown, the user can select the emulation platform directly when using `west build` without an option, because they have full control over the initial cmake cache variables on the west command line.

I still don't see a justification that this use case is going to be needed often enough by users that it merits taking up a west command line option, so please remove it or argue your case :).

File: CMakeLists.txt (hunk 3 above)
Comment (fkokosinski) [reply CMakeLists.txt:1599]: Thanks for spotting that. I've added an additional check to test whether the `EMU_PLATFORM` list contains the value of `EMU_PLATFORM_TARGET`.

File: scripts/pylib/twister/twisterlib.py (hunk 4 above)
Comment (fkokosinski) [reply scripts/pylib/twister/twisterlib.py]: Right. I've dropped this change.

File: boards/riscv/hifive1/board.cmake (hunk 2 above)
Comment (fkokosinski) [reply boards/riscv/hifive1/board.cmake]: Removed the `EMU_PLATFORM_DEFAULT` variable.

File: CMakeLists.txt (hunk 1 above)
Comment (fkokosinski) [reply CMakeLists.txt]: Thanks, I've removed the warning message.

**Comment by fkokosinski** [comment]
@mbolivar-nordic thanks for your insight. I've dropped the west changes.

> Also, please add appropriate documentation for this feature.

@tejlmand I've added documentation in `doc/application` but I understand that it might not be the best place to document this variable.  I've looked for a place where build variables are documented and found only this one. Are you perhaps aware of a better place to document this?

File: CMakeLists.txt [hunk 5]
```diff
@@ -1587,8 +1587,20 @@ if(HEX_FILES_TO_MERGE)
   message(VERBOSE "Merging hex files: ${HEX_FILES_TO_MERGE}")
 endif()
 
-if(EMU_PLATFORM)
-  include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
+if(EMU_PLATFORM OR SUPPORTED_EMU_PLATFORMS)
```
Comment (nashif) [inline CMakeLists.txt:1590]: do we need both EMU_PLATFORM and SUPPORTED_EMU_PLATFORMS? Why not just move to a list type SUPPORTED_EMU_PLATFORMS in general and remove EMU_PLATFORM. Platforms with one supported type will only have 1 value in this case.

File: CMakeLists.txt [hunk 6]
```diff
@@ -1587,8 +1587,20 @@ if(HEX_FILES_TO_MERGE)
...
+  if(NOT EMU_PLATFORM)
+    list(GET SUPPORTED_EMU_PLATFORMS 0 EMU_PLATFORM)
```
Comment (wangnuannuan) [inline CMakeLists.txt:1592]: In twisterlib.py, the default value of simulation is "na", https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/twisterlib.py#L1525. And if cmake choose a default value from `SUPPORTED_EMU_PLATFORMS`, the cmake should give a message to tell user which `EMU_PLATFORM` it choose

File: CMakeLists.txt (hunk 5 above)
Comment (tejlmand) [reply CMakeLists.txt:1590]: Agree, just as proposed here as well: https://github.com/zephyrproject-rtos/zephyr/pull/38102#issuecomment-909150046

File: CMakeLists.txt (hunk 6 above)
Comment (fkokosinski) [reply CMakeLists.txt:1592]: Hi, there was a warning message here initially, but I've got a [review comment](https://github.com/zephyrproject-rtos/zephyr/pull/38102#discussion_r699219226) that it should be removed.

File: scripts/pylib/twister/twisterlib.py [hunk 7]
```diff
@@ -2017,6 +2017,7 @@ def run_cmake(self, args=[]):
         cmake_args = [
             f'-B{self.build_dir}',
             f'-S{self.source_dir}',
+            f'-DEMU_PLATFORM={self.platform.simulation}',
```
Comment (gopiotr) [inline scripts/pylib/twister/twisterlib.py:2020]: If you add this CMake argument `EMU_PLATFORM` to every CMake build running via Twister it enable to run simulator properly to this set in yaml file, as you said in previously comments. But this set CMake argument cause also unexpected CMake warning log in situation, when you run Twister on real target (not on emulaiton). In this case `self.platform.simulation` variable is set as "na" and this cause appear in CMake log misleading information: `CMake file for na not found!`. I propose setting this CMake arguments by `extra_args` in yaml file or eventually (but not recommended) by create if statement in `twisterlib.py` and add this CMake argument only if `self.platform.simulation` is not "na"

**Review by gopiotr — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 5 above)
Comment (fkokosinski) [reply CMakeLists.txt:1590]: Hi, this PR is currently using `EMU_PLATFORM` as a variable that enables choosing between different emulation platforms that are specified for a given board in the `SUPPORTED_EMU_PLATFORMS` list. Removing it would mean that we would need to introduce in this PR other mechanism that would allow such feature. Currently e.g. twister script through yaml file chooses which software to use in the CI by utilizing `EMU_PLATFORM` variable.

File: scripts/pylib/twister/twisterlib.py (hunk 7 above)
Comment (fkokosinski) [reply scripts/pylib/twister/twisterlib.py:2020]: If I understand it correctly, platform schema doesn't define `extra_args` field, so it's not possible for it to be set for each board individually. Or did you mean the `extra_args` as defined by testcase schema? If so, that would mean forcing a testcase to be run on one emulation platform regardless of the board that is being emulated.

Comment (gopiotr) [reply scripts/pylib/twister/twisterlib.py:2020]: You're right, `extra_args` concern only testcase schema - sorry, I omit it. 
So you can consider add `EMU_PLATFORM` in conditional statement like this:
```
if self.platform.simulation != "na":
    cmake_args.append(f'-DEMU_PLATFORM={self.platform.simulation}')
```

Or try change somehow if conditional statement in `CMakeLists.txt` to avoid misleading warning in case of use Twister on real board from there: [CMakeLists.txt](https://github.com/zephyrproject-rtos/zephyr/blob/af7dad518b9902fdb48d32e350c0b920c91192cd/CMakeLists.txt#L1602)

Comment (fkokosinski) [reply scripts/pylib/twister/twisterlib.py]: I've added the conditional statement you suggested. Thanks for spotting the potential problem with this change!

File: scripts/pylib/twister/twisterlib.py (hunk 4 above)
Comment (tejlmand) [reply scripts/pylib/twister/twisterlib.py]: but twister will use the default board per default, which we defined to be the first board in the list.

Which you are doing here:
https://github.com/zephyrproject-rtos/zephyr/blob/7334d3504c1389a61f66ff40fbf12a4b1b66d994/boards/riscv/hifive1/board.cmake#L3

So what is the value in specifying the same emulation software here, as it will be the same as the default:
https://github.com/zephyrproject-rtos/zephyr/blob/7334d3504c1389a61f66ff40fbf12a4b1b66d994/boards/riscv/hifive1/hifive1.yaml#L8

Are there any usecases where the `simulation: <emulator>` in `<board-dir>/<board>.yaml` would be different than the default emulator in `<board-dir>/board.cmake` ?
I would assume we would like twister to use the default emulator, unless explicitly specified in a `testcase.yaml` file.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Did some extra thinking on this, and I think we should completely get rid of the `-DEMU_PLATFORM_TARGET=<emulator>` CMake user setting.

Instead, when defining a list of supported emulator platforms here:
https://github.com/zephyrproject-rtos/zephyr/blob/7334d3504c1389a61f66ff40fbf12a4b1b66d994/boards/riscv/hifive1/board.cmake#L3

then we should loop that list here:
https://github.com/zephyrproject-rtos/zephyr/blob/293af9e8237b28d9ae21037c3ccf48e77861b463/CMakeLists.txt#L1621

and create a run target per emulator.
That would require updates in all emulator cmake files, so that the run and debugserver targets are postfixed, for example like:
https://github.com/zephyrproject-rtos/zephyr/blob/293af9e8237b28d9ae21037c3ccf48e77861b463/cmake/emu/qemu.cmake#L40
becomes:
```
  run_qemu
```

Now ,that will allow the user to do a single CMake invocation, and decide at a later time which emulator to use, for example:
`ninja run_qemu` or `ninja run_renode`.

Now, in order to still support the normal `run` target, we can do:
```
list(GET SUPPORTED_EMU_PLATFORMS 0 default_emu)
add_custom_target(run DEPENDS run_${default_emu})
```

which means that in your example with both `renode` and `qemu` available, a user can do:
```
ninja run_renode  # Will run in the renode emulator (default)
ninja run_qemu    # Will run in the qemu emulator
ninja run         # Will run in the default emulator, which in this case is identical to run_renode
```

**Comment by mbolivar-nordic** [comment]
> becomes:
> 
> ```
>   run_qemu
> ```

I don't know about that @tejlmand.

That would seem to take us even further away from a `west run -r qemu` world, which is what we want, isn't it? Maybe we should resurrect #35366 along those lines instead.

**Comment by tejlmand** [comment]
> That would seem to take us even further away from a `west run -r qemu` world, which is what we want, isn't it? Maybe we should resurrect #35366 along those lines instead.

Would it ?
qemu still creates just a single run target like always, it's just called `run_qemu` instead of `run`.

Meaning that we avoid special CMake flags like: `-DEMU_PLATFORM_TARGET=<emulator>`.

Thinking about it, I actually think that starting to introduce new user flags like `-DEMU_PLATFORM_TARGET=<emulator>` is taking us further away from #35366.

But i'm fully in favor of getting #35366 resurrected, because having emulator execution as build targets in the build system is a misnormer.
The build system should build stuff, and we have `west` as a tool for other task related to development.

But if I short term have to choose between: (long term goal is #35366, )
1. `run_<emulator>` (with `run` mapping to default emulator) as build targets
2. A new `-DEMU_PLATFORM_TARGET=<emulator>` flag

then i'd choose 1. as that is closets to what we have and has a minimal impact when going towards #35366, compared to 2.

**Comment by mbolivar-nordic** [comment]
OK, fair enough.

**Comment by fkokosinski** [comment]
@tejlmand thanks for your comment. I've adapted these changes to the first variant you proposed, i.e. the one creating a separate run target per emulation platform. Dropping the `EMU_PLATFORM` variable completely required changing all `board.cmake` files that were previously using it.

I'm not sure how I should go about documenting this change (as per previous request). I've updated run target mentions in the documentation. The only place that mentions the `EMU_PLATFORM` variable in the documentation is for the [Xtensa sim](https://docs.zephyrproject.org/latest/boards/xtensa/xt-sim/doc/index.html?highlight=emu_platform), but the cmake for it is not provided in the `cmake/emu` directory.

**Comment by fkokosinski** [comment]
@tejlmand @mbolivar could you take a look at this again?

**Comment by tejlmand** [comment]
> @tejlmand @mbolivar could you take a look at this again?

@mbolivar-nordic ^^^

File: CMakeLists.txt [hunk 8]
```diff
@@ -1617,8 +1617,14 @@ if(HEX_FILES_TO_MERGE)
...
 
-if(EMU_PLATFORM)
-  include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
+if(SUPPORTED_EMU_PLATFORMS)
+  foreach(EMU_PLATFORM ${SUPPORTED_EMU_PLATFORMS})
+    if(EXISTS "${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake")
+      include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
+    else()
+      message(WARNING "CMake file for ${EMU_PLATFORM} not found!")
+    endif()
```
Comment (tejlmand) [inline CMakeLists.txt:1626]: Why a warning here ?

The `SUPPORTED_EMU_PLATFORMS` are now specified by the boards `board.cmake` files and is not a user input argument.
So I think this should just fail hard if there is not a corresponding emu platform file.

The biggest issue with CMake warnings is that they don't fail CI builds, meaning someone renaming from `nsim` to `nsim-emu` and not updating  all places will still pass CI.

So I would keep existing behavior of a hard error here:
```suggestion
    include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
```

will result in:
```
CMake Error at /projects/github/ncs/zephyr/CMakeLists.txt:1622 (include):
  include could not find requested file:

    /projects/github/ncs/zephyr/cmake/emu/bad-emulator.cmake
```
that will also fail CI.

Instead of:
```
CMake Warning at /projects/github/ncs/zephyr/CMakeLists.txt:1625 (message):
  CMake file for bad-emulator not found!
```
that will pass CI.

That way we ensure that a supported emulator always have a corresponding cmake file.

File: scripts/pylib/twister/twisterlib.py [hunk 9]
```diff
@@ -1107,7 +1109,7 @@ def handle(self):
 
         logger.debug("Running %s (%s)" % (self.name, self.type_str))
         command = [self.generator_cmd]
-        command += ["-C", self.build_dir, "run"]
+        command += ["-C", self.build_dir, "run_qemu"]
```
Comment (tejlmand) [inline scripts/pylib/twister/twisterlib.py:1112]: why this change ?
The `run` target should still default to the first entry in the `SUPPORTED_EMU_PLATFORMS` list.

Which in this case means it should still be possible to just execute `run`.

File: doc/application/index.rst [hunk 10]
```diff
@@ -965,13 +965,13 @@ hardware. Follow these instructions to run an application via QEMU:
 
    .. code-block:: console
 
-      west build -t run
+      west build -t run_qemu
```
Comment (tejlmand) [inline doc/application/index.rst:968]: Why this change.

Let's keep the info that users can still use `west build -t run` and have the default emulator used.
And then a note that a specific emulator can be chosen with `west build -t run_<emulator>`.

File: doc/application/index.rst [hunk 11]
```diff
@@ -965,13 +965,13 @@ hardware. Follow these instructions to run an application via QEMU:
...
 
    or
 
    .. code-block:: console
 
-      ninja run
+      ninja run_qemu
```
Comment (tejlmand) [inline doc/application/index.rst:974]: Why this change.

Let's keep the info that users can still use `ninja run` and have the default emulator used.
And then a note that a specific emulator can be chosen with `ninja run_<emulator>`.

File: doc/guides/beyond-GSG.rst [hunk 12]
```diff
@@ -303,7 +303,7 @@ Next, run the application.
 
 .. code-block:: console
 
-   west build -t run
+   west build -t run_native
```
Comment (tejlmand) [inline doc/guides/beyond-GSG.rst:306]: Again, I don't think native will have any other emulator supported than native itself.

So being able to still just type `west build -t run` makes things much simpler for the ordinary use-case.

**Review by tejlmand — CHANGES_REQUESTED** [review]
You missed this part:
https://github.com/zephyrproject-rtos/zephyr/pull/38102#pullrequestreview-749604727

> Now, in order to still support the normal run target, we can do:
>
> ```
> list(GET SUPPORTED_EMU_PLATFORMS 0 default_emu)
> add_custom_target(run DEPENDS run_${default_emu})
> ```
> which means that in your example with both renode and qemu available, a user can do:
>
> ```
> ninja run_renode  # Will run in the renode emulator (default)
> ninja run_qemu    # Will run in the qemu emulator
> ninja run         # Will run in the default emulator, which in this case is identical to run_renode
> ```

we should still support the existing `run` etc. targets.
Having the ability to specify an exact emulator target is a nice addition, but let's still support the existing behavior.

File: CMakeLists.txt (hunk 8 above)
Comment (fkokosinski) [reply CMakeLists.txt]: Done

**Comment by fkokosinski** [comment]
@tejlmand thanks for you comment. I've reintroduced the `run` target as you suggested, which runs the first entry in the `SUPPORTED_EMU_PLATFORMS` list.

File: cmake/emu/qemu.cmake [hunk 13]
```diff
@@ -37,8 +37,8 @@ if(CONFIG_QEMU_UEFI_BOOT)
 endif()
 
 set(qemu_targets
-  run
-  debugserver
+  run_qemu
+  debugserver_qemu
```
Comment (tejlmand) [inline cmake/emu/qemu.cmake:41]: Please remember to update all occurrences of debugserver.
The `debugserver_qemu` no longer waits for a connections, because this line is not updated:
 https://github.com/zephyrproject-rtos/zephyr/blob/f397e9b6c2e5268009e303210807d4b9045e1eb4/cmake/emu/qemu.cmake#L337

Note: there could be more places in need of an update, so please do a thorough search.

File: CMakeLists.txt [hunk 14]
```diff
@@ -1622,8 +1622,13 @@ if(HEX_FILES_TO_MERGE)
   message(VERBOSE "Merging hex files: ${HEX_FILES_TO_MERGE}")
 endif()
 
-if(EMU_PLATFORM)
-  include(${ZEPHYR_BASE}/cmake/emu/${EMU_PLATFORM}.cmake)
+if(SUPPORTED_EMU_PLATFORMS)
+  list(GET SUPPORTED_EMU_PLATFORMS 0 default_emu)
+  add_custom_target(run DEPENDS run_${default_emu})
```
Comment (tejlmand) [inline CMakeLists.txt:1627]: Not directly related to this line, but didn't know where else to put the following comment.

We need something similar for the `debugserver`, but as not all emulators has the `debugserver` target today then it should probably be something along:
```
if(TARGET debugserver_${default_emu})
  add_custom_target(debugserver DEPENDS debugserver_${default_emu})
endif()
```

so that only if the default emulator supports debugserver then we create that target.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Thanks for the update, looks much cleaner.

Still some small things to be fixed, see specific comments.

Also this was missed from my comments:
https://github.com/zephyrproject-rtos/zephyr/pull/38102#discussion_r718380151
https://github.com/zephyrproject-rtos/zephyr/pull/38102#discussion_r718380650
> And then a note that a specific emulator can be chosen with west build -t run_<emulator>.

We must inform the users about this new feature, and how to choose a specific emulator when multiple emulators are supported.

So I unresolved those comments.

File: cmake/emu/qemu.cmake (hunk 13 above)
Comment (fkokosinski) [reply cmake/emu/qemu.cmake:41]: Thanks for spotting that, fixed. I didn't find any other places that needed to be updated, so I'm resolving this conversation.

File: CMakeLists.txt (hunk 14 above)
Comment (fkokosinski) [reply CMakeLists.txt:1627]: Done. This also helped me spot that I was missing the change from `EMU_PLATFORM` to `SUPPORTED_EMU_PLATFORMS` in `cmake/flash/CMakeLists.txt`.

**Comment by fkokosinski** [comment]
@tejlmand thanks for your suggestions. I've reintroduced the `debugserver` target, done the fixes regarding new target names and added a note to the docs. Please let me know if I've missed something.

**Review by tejlmand — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by carlescufi** [comment]
@tejlmand and @gopiotr could you take another look please?

**Comment by carlescufi** [comment]
@gopiotr @nashif could you take another look?

**Review by nashif — APPROVED** [review]
